helpers/core.js

```diff
@@ -78,6 +92,15 @@ var helpers = {
 						aliases[asVariable] = item;
 					}
 
+					if (!isEmptyObject(hashOptions)) {
+						if (hashOptions.value) {
+							aliases[hashOptions.value] = item;
+						}
+						if (hashOptions.index) {
+							aliases[hashOptions.index] = index;
+						}
+					}
```

Comment:
It probably wasn't clear from the issue, but we also want to replace using `as` with objects. So you should be able to do `{{#each props propValue=value prop=key}}`.

So, you can use `value`/`index` with arrays
and `value`/`key` with objects